src/test/java/hudson/plugins/jira/JiraProjectPropertyTest.java

```diff
@@ -0,0 +1,131 @@
+package hudson.plugins.jira;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import hudson.model.FreeStyleProject;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class JiraProjectPropertyTest {
+
+    @Rule
+    public final JenkinsRule r = new JenkinsRule();
+
+    private FreeStyleProject freeStyleProject;
+    private Folder folder;
+    private List<JiraSite> firstList;
+
+    @Before
+    public void initialize() throws Exception {
+        folder = r.jenkins.createProject(Folder.class, "first");
+        JiraFolderProperty jiraFolderProperty = new JiraFolderProperty();
+        firstList = new ArrayList<>();
+        firstList.add(new JiraSite("https://first.com/"));
+        jiraFolderProperty.setSites(firstList);
+        folder.getProperties().add(jiraFolderProperty);
+    }
+
+    @Test
+    public void getSitesNullWithoutFolder() throws Exception {
+        FreeStyleProject freeStyleProject = r.createFreeStyleProject();
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        freeStyleProject.addProperty(prop);
+        JiraSite site = prop.getSite();
+        assertNull(site);
+        JiraProjectProperty actual = freeStyleProject.getProperty(JiraProjectProperty.class);
+        assertNotNull(actual);
+        assertNull(actual.getSite());
+    }
+
+    @Test
+    public void getSitesNullWithFolder() throws Exception {
+        freeStyleProject = folder.createProject(FreeStyleProject.class, "something");
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        freeStyleProject.addProperty(prop);
+        JiraProjectProperty property = freeStyleProject.getProperty(JiraProjectProperty.class);
+        assertNotNull(property);
+        assertNull(property.getSite());
+    }
+
+    @Test
+    public void testNoInitialDefault() throws Exception {
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        setupGlobalConfig("single-site.yml");
+        JiraSite site = prop.getSite();
+        assertNotNull(site);
+        assertEquals("https://jira.com/", site.getName());
+    }
+
+    @Test
+    public void testDefault() throws Exception {
```

Comment:
please avoid using word "test" in your test names (it doesn't add any useful information, we already know it's a test)
please give tests more descriptive names

Reply:
There I avoided the name test, I went ahead and removed it from all existing tests.
I will fix your other comments once I get time.